### PR TITLE
#10681: Add perf benchmark support to Mixtral test_perf and demo

### PR DESCRIPTION
**Pull request by mtairum**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/10681

### Problem description
Add benchmark profile to Mixtral perf test and to demo code.

### Checklist
- [x] Test_perf support
- [x] Demo_with_prefill support

Tests:
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/10213500335)
- [x] [T3k Perf pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/10213495090)
- [x] [T3k demo pipeline](https://github.com/tenstorrent/tt-metal/actions/runs/10213504451)

### CODE REVIEW

**Comment by mtairum** [comment]
Added benchmark support to perf test. 

Missing `demo_with_prefill.py` now.

**Comment by mtairum** [comment]
@ttmchiou What are your thoughts in the changes in file `.github/workflows/t3000-demo-tests.yaml`?

At the time, only Falcon7B and Mixtral have support for the benchmark profile. But as we ramp up more models we will include the pass that searches for the benchmark files and uploads if found to all our models.

**Comment by mtairum** [comment]
Pipeline is correctly uploading the benchmark csv to the dashboard: https://github.com/tenstorrent/tt-metal/actions/runs/10185641669/job/28175992490

Re-running the relevant tests again to prepare for merge.

File: models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py [hunk 1]
```diff
@@ -294,6 +349,85 @@ def run_mixtral_demo(user_input, batch_size, device_mesh, instruct_mode, is_ci_e
...
+    logger.info(f"Prefill inference time per user: {round(measurements['inference_prefill']/(batch_size-1), 4)}s")
+    logger.info(
+        f"Total Decode inference time ({max_generated_tokens-1} iterations): {round(measurements['inference_decode'], 4)}s"
+    )
+    logger.info("---")
+    logger.info(f"Time to first token: {round(measurements['prefill_time_to_token'], 4) * 1000}ms")
+    logger.info(f"Average tokens/sec/user: {round(measurements['decode_t/s/u'], 2)}")
+
+    target_prefill_ts = 5000  # TODO Update prefill target
+    target_decode_ts = 1056
```
[the remark this answers is not in the document]
Comment (mtairum) [reply models/demos/t3000/mixtral8x7b/demo/demo_with_prefill.py:406]: Yes, that's the correct target for decode: 33 tok/s/u (showed in main page) in a batch of 32 users will be 1056 tok/s.
The prefill target we haven't properly calculated it yet.

**Review by uaydonat — APPROVED** [review]

**Review by ttmchiou — APPROVED** [review]

**Pull request merged**